docs/mosaicking.rst

```diff
@@ -0,0 +1,310 @@
+all the inpute images. Note that you don't necessarily need to use this if you
+already know the final header or WCS you want to use for the images - in this
+case you can skip straight to :ref:`coadding`.
+
+You can optionally provide options to try and constrain
+the solution, as we will see below. To start off, let's consider the simplest
+example, which is to call :func:`~reproject.mosaicking.find_optimal_celestial_wcs`
+with the files downloaded above, but no additional information::
+
+    >>> from reproject.mosaicking import find_optimal_celestial_wcs
+    >>> wcs_out, shape_out = find_optimal_celestial_wcs(m17_hdus)
+
+The first argument to :func:`~reproject.mosaicking.find_optimal_celestial_wcs`
+should be a list where each element is either a filename, an HDU object (e.g.
+`~astropy.io.fits.PrimaryHDU` or `~astropy.io.fits.ImageHDU`), an
+`~astropy.io.fits.HDUList` object, or a tuple of ``(array, wcs)``. In the
+example above, we have passed a list of HDUs. We can now look at the output
+WCS and shape::
+
+    >>> wcs_out.to_header()  # doctest: +FLOAT_CMP
+    WCSAXES =                    2 / Number of coordinate axes
+    CRPIX1  =      900.07981909507 / Pixel coordinate of reference point
+    CRPIX2  =      1099.9484609446 / Pixel coordinate of reference point
+    CDELT1  =     -0.0002777777845 / [deg] Coordinate increment at reference point
+    CDELT2  =      0.0002777777845 / [deg] Coordinate increment at reference point
+    CUNIT1  = 'deg'                / Units of coordinate increment and value
+    CUNIT2  = 'deg'                / Units of coordinate increment and value
+    CTYPE1  = 'RA---TAN'           / Right ascension, gnomonic projection
+    CTYPE2  = 'DEC--TAN'           / Declination, gnomonic projection
+    CRVAL1  =      275.18472258448 / [deg] Coordinate value at reference point
+    CRVAL2  =     -16.141349044263 / [deg] Coordinate value at reference point
+    LONPOLE =                180.0 / [deg] Native longitude of celestial pole
+    LATPOLE =     -16.141349044263 / [deg] Native latitude of celestial pole
+    RADESYS = 'FK5'                / Equatorial coordinate system
+    EQUINOX =               2000.0 / [yr] Equinox of equatorial coordinates
+    >>> shape_out
+    (2201, 1800)
+
+Coordinate system
+-----------------
+
+By default, the coordinate system of the first file is used, and the final
+WCS is set up so that North (in that coordinate system) is up. In the
+case above, the images are in equatorial coordinats, so the final WCS is also
+in equatorial coordinates. We can force the output WCS to instead be in
+Galactic coordinates by setting the ``frame=`` argument to a coordinate frame
+object such as :class:`astropy.coordinates.Galactic`::
+
+    >>> from astropy.coordinates import Galactic
+    >>> wcs_out, shape_out = find_optimal_celestial_wcs(m17_hdus,
+    ...                                                 frame=Galactic())
```

Comment:
do strings work?  e.g., `'fk5'`, `'icrs'`, `'galactic'`?  For consistence with the `astropy.coordinates` API, they should

Reply:
Fixed